runtime/lua/editorconfig.lua

```diff
@@ -0,0 +1,258 @@
+    'trim_trailing_whitespace must be either "true" or "false"'
+  )
+  if val == 'true' then
+    vim.api.nvim_create_autocmd('BufWritePre', {
+      group = 'editorconfig',
+      buffer = bufnr,
+      callback = function()
+        local view = vim.fn.winsaveview()
+        vim.api.nvim_command('silent! undojoin')
+        vim.api.nvim_command('silent keepjumps keeppatterns %s/\\s\\+$//e')
+        vim.fn.winrestview(view)
+      end,
+    })
+  else
+    vim.api.nvim_clear_autocmds({
+      event = 'BufWritePre',
+      group = 'editorconfig',
+      buffer = bufnr,
+    })
+  end
+end
+
+function M.properties.insert_final_newline(bufnr, val)
+  assert(val == 'true' or val == 'false', 'insert_final_newline must be either "true" or "false"')
+  vim.bo[bufnr].fixendofline = val == 'true'
+  vim.bo[bufnr].endofline = val == 'true'
+end
+
+--- Modified version of |glob2regpat()| that does not match path separators on *.
+---
+--- This function replaces single instances of * with the regex pattern [^/]*. However, the star in
+--- the replacement pattern also gets interpreted by glob2regpat, so we insert a placeholder, pass
+--- it through glob2regpat, then replace the placeholder with the actual regex pattern.
+---
+---@param glob string Glob to convert into a regular expression
+---@return string Regular expression
+---
+---@private
+local function glob2regpat(glob)
+  local placeholder = '@@PLACEHOLDER@@'
+  return (string.gsub(
+    vim.fn.glob2regpat(
+      vim.fn.substitute(
+        string.gsub(glob, '{(%d+)%.%.(%d+)}', '[%1-%2]'),
+        '\\*\\@<!\\*\\*\\@!',
+        placeholder,
+        'g'
+      )
+    ),
+    placeholder,
+    '[^/]*'
+  ))
+end
+
+--- Normalize path separators
+---
+---@param path string Path
+---@return string Normalized path
+---
+---@private
+local function convert_pathseps(path)
+  if is_win then
+    return (path:gsub('\\', '/'))
+  end
+
+  return path
+end
+
+--- Parse a single line in an EditorConfig file
+---
+---@param line string Line
+---@return string|nil If the line contains a pattern, the glob pattern
+---@return string|nil If the line contains a key-value pair, the key
+---@return string|nil If the line contains a key-value pair, the value
+---
+---@private
+local function parse_line(line)
+  if line:find('^%s*[^ #;]') then
+    local glob = (line:match('%b[]') or ''):match('^%s*%[(.*)%]%s*$')
+    if glob then
+      return glob, nil, nil
+    end
+
+    local key, val = line:match('^%s*([^:= ][^:=]-)%s*[:=]%s*(.-)%s*$')
+    if key ~= nil and val ~= nil then
+      return nil, key:lower(), val:lower()
+    end
+  end
+end
+
+--- Parse options from an .editorconfig file
+---
+---@param filepath string File path of the file to apply EditorConfig settings to
+---@param dir string Current directory
+---@return table Table of options to apply to the given file
+---
+---@private
+local function parse(filepath, dir)
+  local pat = nil
+  local opts = {}
+  local f = io.open(dir .. '/.editorconfig')
```

Comment:
Unconverted pathsep

Reply:
Good catch. `editorconfig.nvim` has had other Windows related bug reports in the past, but not in a while, and never about this. So I think that means it just works on Windows, but I'll try and find out for sure.

Reply:
Windows supports both / and \ paths.